odoo/orm/models.py

```diff
@@ -5370,10 +5375,11 @@ def _apply_ir_rules(self, query: Query, mode: str = 'read') -> None:
             return
 
         # apply main rules on the object
-        Rule = self.env['ir.rule']
-        domain = Rule._compute_domain(self._name, mode)
+        domain = self.env['ir.rule']._compute_domain(self._name, mode)
         if domain:
```

Comment:
For the sake of clarity, I suggest to more explicit:
```py
if not domain.is_true():
```
like you did in `_where_calc`